packages/agent/src/utils/oid4vciCredentialSuppliers.ts

```diff
@@ -242,18 +252,66 @@ const supplierByType = async (args: CredentialDataSupplierArgs): Promise<Credent
     }
 }
 
-export function getCredentialDataSupplier(id: string): CredentialDataSupplier {
-    let supplier: CredentialDataSupplier
-    if (id.match(/(future)|(fma2023)|(fmdm2023)/)) {
-        supplier = credentialDataSupplierFMAGuest2023
-    } else if (id.match(/(dbc)|(blockchain)/)) {
-        supplier = credentialDataSupplierDBCConference2023
-    } else if (id.match(/(triall)|(cix)/)) {
-        supplier = credentialDataSupplierTriallGuest2023
-    } else if (id.match(/(energy)/)) {
-        supplier = credentialDataSupplierEnergySHRGuest2023
-    } else {
-        supplier = supplierByType
+
+export function getCredentialDataSupplier(issuerCorrelationId: string): CredentialDataSupplier {
+    const templateCredentialDataSupplier = new TemplateCredentialDataSupplier(issuerCorrelationId)
+    return templateCredentialDataSupplier.generateCredentialData.bind(templateCredentialDataSupplier)
+}
+
+class TemplateCredentialDataSupplier {
+    private readonly issuerCorrelationId: string
+
+    constructor(correlationId: string) {
+        this.issuerCorrelationId = correlationId
+    }
+
+    public async generateCredentialData(args: CredentialDataSupplierArgs): Promise<CredentialDataSupplierResult> {
+        const hardCodedSupplierResult = await this.getHardCodedSupplierCredential(this.issuerCorrelationId, args)
+        if (hardCodedSupplierResult.credential) {
+            return Promise.resolve(hardCodedSupplierResult)
+        }
+
+        const types: string[] = getTypesFromRequest(args.credentialRequest)
+        const credentialSupplierConfig = args.credentialSupplierConfig as CredentialSupplierConfigWithTemplateSupport
+        if (credentialSupplierConfig.template_mappings) {
+            const templateMapping = credentialSupplierConfig.template_mappings
+                .find(mapping => mapping.credential_types.some(type => types.includes(type)))
+            if (templateMapping) {
+                const baseDir = credentialSupplierConfig?.templates_base_dir ? `/${credentialSupplierConfig.templates_base_dir}` : '';
+                const templatePath = `${CONF_PATH}${baseDir}/${templateMapping.template_path}`;
```

Comment:
I rather see a bit of logic looking at slashes when concatenating. You don't want to end up with 'folder1//folder2' or 'folder1folder2`. There are 3 vars here so it could go wrong for all three of them

Reply:
Added function normalizeFilePath